scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -277,22 +277,15 @@ def run_build(self, args=[]):
     def run_cmake(self, args=""):
 
         if not self.options.disable_warnings_as_errors:
-            ldflags = "-Wl,--fatal-warnings"
-            cflags = "-Werror"
-            aflags = "-Werror -Wa,--fatal-warnings"
-            gen_defines_args = "--edtlib-Werror"
+            force_warnings_as_error = 1
         else:
-            ldflags = cflags = aflags = ""
-            gen_defines_args = ""
+            force_warnings_as_error = 0
 
         logger.debug("Running cmake on %s for %s" % (self.source_dir, self.platform.name))
         cmake_args = [
             f'-B{self.build_dir}',
             f'-DTC_RUNID={self.instance.run_id}',
-            f'-DEXTRA_CFLAGS={cflags}',
-            f'-DEXTRA_AFLAGS={aflags}',
-            f'-DEXTRA_LDFLAGS={ldflags}',
-            f'-DEXTRA_GEN_DEFINES_ARGS={gen_defines_args}',
+            f'-DFORCE_WARNING_AS_ERRORS={force_warnings_as_error}',
```

Comment:
```suggestion
            f'-DFORCE_WARNING_AS_ERRORS={1 if self.options.disable_warnings_as_errors else 0}',
```

Reply:
I returned `gen_defines_args` back, so I decided to leave original version with `if-else` construction.

Reply:
In that case we need to also remove `#FIXME: gen_defines_args = "--edtlib-Werror"`